router/core/graphql_handler.go

```diff
@@ -159,12 +169,43 @@ func (h *GraphQLHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
-		if sharedPreparedPlan == nil {
-			requestLogger.Error("prepare plan is nil", zap.Error(err))
+		preparedPlan, ok = sharedPreparedPlan.(planWithMetaData)
+		if !ok {
+			requestLogger.Error("unexpected prepared plan type")
 			w.WriteHeader(http.StatusInternalServerError)
+			writeRequestErrors(graphql.RequestErrorsFromError(internalServerErrorErr), w, requestLogger)
 			return
 		}
-		preparedPlan = sharedPreparedPlan.(planWithExtractedVariables)
+	}
+
+	if h.gqlMetricsExporter != nil {
+		fieldUsageInfos := make([]*graphqlmetricsv1.TypeFieldUsageInfo, len(preparedPlan.schemaUsageInfo.TypeFields))
+
+		for i := range preparedPlan.schemaUsageInfo.TypeFields {
+			fieldUsageInfos[i] = &graphqlmetricsv1.TypeFieldUsageInfo{
+				Count:     1,
+				Path:      preparedPlan.schemaUsageInfo.TypeFields[i].Path,
+				TypeNames: preparedPlan.schemaUsageInfo.TypeFields[i].TypeNames,
+				SourceIDs: preparedPlan.schemaUsageInfo.TypeFields[i].Source.IDs,
+			}
+		}
+
+		h.gqlMetricsExporter.Record(&graphqlmetricsv1.SchemaUsageInfo{
```

Comment:
At this point, I would only register the Hash and the Document.

Reply:
As discussed. Let's keep super optimizations out of the initial version. At the collector and database level, we already ensure that operations aren't stored for every request. The exporter won't send the requested document for the same usage twice but for every batch which is done by default every 10s or every 512 requests in batches. This makes the current implementation very simple because we don't need to coordinate it with the collector.